Sources/MapboxNavigation/StyleManager.swift

```diff
@@ -193,18 +210,38 @@ open class StyleManager {
         forceRefreshAppearance()
     }
     
-    // Workaround to refresh appearance by removing all views and then adding them again.
-    // UITextEffectsWindow will be created when system keyboard is shown and cannot be safely removed.
     func forceRefreshAppearance() {
-        for window in UIApplication.shared.windows {
-            if !window.isKind(of: NSClassFromString("UITextEffectsWindow") ?? NSString.classForCoder()) {
-                for view in window.subviews {
-                    view.removeFromSuperview()
-                    window.addSubview(view)
+        // Use trait collection to detect what window should be updated.
+        if #available(iOS 13.0, *) {
+            UIApplication.shared.connectedScenes.forEach {
+                if let windowScene = $0 as? UIWindowScene,
+                   traitCollection?.userInterfaceIdiom == .phone {
+                    refreshAppearance(for: windowScene.windows)
+                } else if let templateApplicationScene = $0 as? CPTemplateApplicationScene,
+                          traitCollection?.userInterfaceIdiom == .carPlay {
+                    let window = templateApplicationScene.carWindow
+                    refreshAppearance(for: [window])
                 }
             }
+        } else {
+            refreshAppearance(for: UIApplication.shared.windows)
         }
         
         delegate?.styleManagerDidRefreshAppearance(self)
     }
+    
+    func refreshAppearance(for windows: [UIWindow]) {
+        // Workaround to refresh appearance by removing all views and then adding them again.
+        // UITextEffectsWindow will be created when system keyboard is shown and cannot be safely removed.
```

Comment:
Nit: This is basically a documentation comment.

```suggestion
    /**
     Workaround to refresh appearance by removing all views and then adding them again.
     UITextEffectsWindow will be created when system keyboard is shown and cannot be safely removed.
     */
    func refreshAppearance(for windows: [UIWindow]) {
```